### DISCUSSION TITLE
Edge starting at top left corner of nodes

### DISCUSSION THREAD

**Discussion by borishenicker** [bcakmakoglu/vue-flow#724] [Q&A] [answered]
Hi,

First thanks for sharing this really nice library !

I've built an app where edges get added automatically when you add a node :

![printscreen](https://user-images.githubusercontent.com/35304221/222024478-00892dea-f91a-4384-852d-2fdd3ac08cf3.png)

From node 1 you have a menu to add node 2, and the edge is added automatically. To do that I am only adding / removing nodes and edges on / from 2 refs `nodes.value` and `edges.value`

I've recently upgraded from 0.4 to the latest 1.15. Before the upgrade the edges where starting at the bottom of the source and ending at the top of the target, but now they seems to start / end at position 0.

I've tried using `addNodes `/ `addEdges`, adding handles and changing the `sourcePosition `and `targetPosition` :

      nodes.value = [
       {
          id: '1',
          type: 'customaction',
          position: { x: 600, y: 100 },
          sourcePosition: Position.Bottom,
          targetPosition: Position.Top,
          class: 'customactionnode-container',
          height: 80,
          width: 150,
        },
        {
          id: '2',
          type: 'customaction',
          position: { x: 600, y: 250 },
          sourcePosition: Position.Bottom,
          targetPosition: Position.Top,
          class: 'customactionnode-container',
          height: 80,
          width: 150,
        },
      ]
      edges.value = [
        {
          id: 'e1-2',
          source: '1',
          target: '2',
          type: 'step',
          class: 'customedge-container',
        },
      ]

If anybody has an idea where this might be coming from. 

Many thanks,

**Comment by bcakmakoglu** [comment, marked as answer]
A codesandbox with a reproduction would be helpful to analyze your case, otherwise it's really difficult to know what is going on :/

**Reply by borishenicker** [reply]
Thanks for your answer, I'll try to reproduce in a sandbox and let you know

**Reply by borishenicker** [reply]
I'm having a hard time recreating in a sandbox. However after testing different versions, I found out that I don't have the issue before 1.6.4. Any chance to retrace the cause from that ?

**Reply by bcakmakoglu** [reply]
Any errors/warnings in the console?
What's your package.json look like?
Possibly got a repo I can look at or something?

I can't really identify any specific change that occurred between 1.6.4 and 1.7.0 that would cause this :(

**Reply by borishenicker** [reply]
No errors / warnings

I can't really share anything since it's in a private repo at the moment. The change lies between 1.6.3 and 1.6.4 (1.6.3 works fine, 1.6.4 doesn't). Still working on the codesandbox in the meantime.

**Reply by borishenicker** [reply]
I've managed to get the same issue in this sandbox : https://codesandbox.io/p/sandbox/loving-varahamihira-n0ioso

Tested with 1.6.3, no issue. I've got it with > 1.6.3

**Reply by bcakmakoglu** [reply]
Thanks for the sandbox :)
After playing around I have identified an issue that seems to be causing this.
It's the `width` and `height` properties in the initial node.

The sandbox seems to work fine if I instead use `style: { width: "150px", height: "80px" },`.

Can you check if your concrete implementation has the same issue? Does using style instead resolve it?

**Reply by borishenicker** [reply]
I swear I tested every single settings BUT changing those 2 :D. Thanks for pointing it out, that indeed solved the issue even with the latest release.

I don't know if it's coming from a doc from an older release or simply a mistake that worked up until now, but that's fine now.

**Reply by bcakmakoglu** [reply]
Though you *shouldn't* have to change your code, it is a bug and I have opened a PR to fix that.
I will publish it in a couple mins so please install `1.15.4` and keep your code as-is and see if that solves it as well :)

**Reply by borishenicker** [reply]
I will let you know

**Reply by bcakmakoglu** [reply]
`1.15.4` is published now

**Reply by borishenicker** [reply]
Many thanks, I don't have the issue anymore in 1.15.4 with `width` and `height `properties in the initial node

**Reply by bcakmakoglu** [reply]
Nice 👍 Thanks for reporting the issue :)
